cub/util_device.cuh

```diff
@@ -158,28 +222,81 @@ CUB_RUNTIME_FUNCTION __forceinline__ cudaError_t PtxVersion(int &ptx_version)
 #endif
 }
 
+/**
+ * \brief Retrieves the PTX version that will be used on \p device (major * 100 + minor * 10).
+ */
+CUB_RUNTIME_FUNCTION __forceinline__ cudaError_t PtxVersionUncached(int &ptx_version, int device)
+{
+    SwitchDevice sd(device);
```

Comment:
This makes twice as many `cudaSetDevice` calls as needed, doesn't it? Because effectively it jumps back and forth, instead of going to 0, 1, 2, 3, ..., and only then back to the original one?

I'm not sure if this is significant, just pointing that out.